hazelcast/config.py

```diff
@@ -1290,6 +1298,43 @@ def fail_on_indeterminate_operation_state(self, value):
         else:
             raise TypeError("fail_on_indeterminate_operation_state must be a boolean")
 
+    @property
+    def creds_username(self):
+        # type: (_Config) -> str
+        return self._creds_username
+
+    @creds_username.setter
+    def creds_username(self, username):
+        # type: (_Config, str) -> None
+        if not isinstance(username, six.string_types):
+            raise TypeError("creds_password must be a string")
+        self._creds_username = username
+
+    @property
+    def creds_password(self):
+        # type: (_Config) -> str
+        return self._creds_password
+
+    @creds_password.setter
+    def creds_password(self, password):
+        # type: (_Config, str) -> None
+        if not isinstance(password, six.string_types):
+            raise TypeError("creds_password must be a string")
+        self._creds_password = password
+
+    @property
+    def token_provider(self):
+        # type: (_Config) -> TokenProvider
+        return self._token_provider
+
+    @token_provider.setter
+    def token_provider(self, token_provider):
+        # type: (_Config, TokenProvider) -> None
+        token_fun = getattr(token_provider, "token", None)
+        if token_fun is None or not isinstance(token_fun, types.MethodType):
+            raise TypeError("token_provider must be an object with a token method")
```

Comment:
I see that we accept anything with a `token()` method but I feel like we should make use of the `TokenProvider` class.

Something like
```suggestion
        if not isinstance(token_provider, TokenProvider):
            raise TypeError("token_provider must be an instance of TokenProvider")
```

Reply:
Since Kerberos token provider is in its own package, that creates a chicken-egg problem ATM. We can do that after releasing 4.2.2.

Reply:
^ I mean 4.2.1

Reply:
Won't the Kerberos package depend on the Hazelcast client 4.2.1+?

Reply:
It doesn't ATM. IMO there's not much need to add a dependency but we may discuss that after we release 4.2.1.

Reply:
OK. BTW, where is the code for it? I haven't seen it